File: app/src/main/java/com.example.safehome/Sensors/DoorSensor.java
package com.example.safehome.Sensors;

import java.util.List;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;


public class DoorSensor implements Sensor {
	private static final Logger log = LoggerFactory.getLogger(DoorSensor.class);
	private List<SensorMomento>[] sensorMomentoList;
	private SensorInfo state;

	/**
	 * detects whether an event has occurred
	 */
	@Override
	public boolean isEventDectected() {
		return false;
	}

	/**
	 * identifies the type of sensor
	 */
	@Override
	public void identify() {

	}

	/**
	 * enables notifications
	 */
	@Override
	public void enableNotification() {

	}

	/**
	 * disables notifications
	 */
	@Override
	public void disableNotification() {

	}

	/**
	 * reconfigures sensor
	 */
	@Override
	public void reconfigure() {

	}

	/**
	 * adds sensor to system
	 */
	@Override
	public void addSensorMomento() {

	}

	/**
	 * gets existing sensor from system
	 */
	@Override
	public void getSensorMomento() {

	}
}

File: app/src/main/kotlin/com.example.safehome/Sensors/DoorSensor.kt
package com.example.safehome.Sensors

import org.slf4j.LoggerFactory

class DoorSensor : Sensor<Any?> {
    private val sensorMomentoList: Array<List<SensorMomento>>
    private val state: SensorInfo? = null

    /**
     * detects whether an event has occurred
     */
    override val isEventDectected: Boolean
        get() = false

    /**
     * identifies the type of sensor
     */
    override fun identify() {}

    /**
     * enables notifications
     */
    override fun enableNotification() {}

    /**
     * disables notifications
     */
    override fun disableNotification() {}

    /**
     * reconfigures sensor
     */
    override fun reconfigure() {}

    /**
     * adds sensor to system
     */
    override fun addSensorMomento() {}

    /**
     * gets existing sensor from system
     */
    override val sensorMomento: Unit
        get() {}

    companion object {
        private val log = LoggerFactory.getLogger(DoorSensor::class.java)
    }
}